### PR TITLE
Fix unique_ptrs

### PR DESCRIPTION
**Pull request by JoshuaSBrown**
Fix memory leaks part 4

### CODE REVIEW

File: include/votca/csg/topology.h [hunk 1]
```diff
@@ -59,7 +60,7 @@ using InteractionContainer = std::vector<Interaction *>;
 class Topology {
  public:
   /// constructor
-  Topology() { _bc = new OpenBox(); }
+  Topology() { _bc = std::unique_ptr<BoundaryCondition>(new OpenBox()); }
```
Comment (JensWehner) [inline include/votca/csg/topology.h]: make unique

File: include/votca/csg/topology.h [hunk 2]
```diff
@@ -271,19 +272,15 @@ class Topology {
...
     }
 
-    if (_bc) {
-      delete (_bc);
-    }
-
     switch (boxtype) {
       case BoundaryCondition::typeTriclinic:
-        _bc = new TriclinicBox();
+        _bc = std::unique_ptr<BoundaryCondition>(new TriclinicBox());
```
Comment (JensWehner) [inline include/votca/csg/topology.h]:
```suggestion
        _bc = std::unique_ptr<TriclinicBox>();
```

File: include/votca/csg/topology.h [hunk 3]
```diff
@@ -300,7 +297,7 @@ class Topology {
    * @brief Return the boundary condition object
    */
   const BoundaryCondition &getBoundary() const {
-    assert(_bc != nullptr && "Cannot return boundary condition is null");
+    assert(_bc && "Cannot return boundary condition is null");
```
Comment (JensWehner) [inline include/votca/csg/topology.h:300]: nice

File: src/libcsg/topology.cc [hunk 4]
```diff
@@ -85,10 +80,7 @@ void Topology::Cleanup() {
     _interactions.clear();
   }
   // cleanup _bc object
-  if (_bc) {
-    delete (_bc);
-  }
-  _bc = new OpenBox();
+  _bc = std::unique_ptr<BoundaryCondition>(new OpenBox());
```
Comment (JensWehner) [inline src/libcsg/topology.cc]: see above

**Review by JensWehner — CHANGES_REQUESTED** [review]
use make_unique 
otherwise good work

File: include/votca/csg/topology.h (hunk 2 above)
Comment (JoshuaSBrown) [reply include/votca/csg/topology.h]: lol this is broken.

Comment (JensWehner) [reply include/votca/csg/topology.h]: `std::make_unique<TriclinicBox>()`

sorry yesterday I was not at my best.

**Comment by junghans** [comment]
@JensWehner please review again!

**Review by JensWehner — APPROVED** [review]
good work

**Pull request merged**